kratos/python/add_mesh_to_python.cpp

```diff
@@ -515,7 +515,12 @@ void  AddMeshToPython(pybind11::module& m)
 //     .def(SolutionStepVariableIndexingPython<Element, Variable<vector<double> > >())
 //     .def(SolutionStepVariableIndexingPython<Element, Variable<DenseMatrix<double> > >())
     .def("Initialize", &EntityInitialize<Element>)
-    //.def("CalculateLocalSystem", &Element::CalculateLocalSystem)
+    .def("EquationIdVector", [](const Element& self, const ProcessInfo& rProcessInfo){
```

Comment:
so `ublas::vector` is passed by copy but `ublas::matrix` is passed by reference?

Reply:
i don't follow your comment...

EquationIdsVector is returned not passed

Reply:
i don't follow you. EquationIds are RETURNED...

Reply:
I mean why do you expose this one differently from `CalculateLocalSystem`
=> passing by ref would be more efficient and consistent

Reply:
well, i thought it is easier to write

~~~py
ids = EquationIdVector(ProcessInfo)
~~~

but i could not do the same with the CalculateLocalSystem since it returns two parameters. (i guess it would be possible to do via pybind, but it is definitely not simple)

Reply:
also there appear to be some problem in passing DenseIntVector by reference. Don't ask me what is the problem exactly...

Reply:
> (i guess it would be possible to do via pybind, but it is definitely not simple)

this is actually quite straight forward (FYI): https://github.com/KratosMultiphysics/CoSimIO/blob/v1.0.1/co_sim_io/python/co_sim_io_python.cpp#L32-L55

> also there appear to be some problem in passing DenseIntVector by reference.

ok this is what I thought. I guess the DenseVector is not exposed like the matrix hence it doesn't work. Just wanted to confirm